### PR TITLE
Alerts (create + delete)

### PR DESCRIPTION
**Pull request by ambuznego**
TODO
Want to handle Show page and Update + tests for this new controller in separate PR

### CODE REVIEW

File: assets/js/actions/alert.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        alert: alert
+      })
+      .then(response => {})
+  }
+}
+
+export const deleteAlert = (id) => {
+  return (dispatch) => {
+
+    return rest.post('/api/alerts/delete', {
```
Comment (vicmgs) [inline assets/js/actions/alert.js:16]: we can make this a delete action right? instead of a post?

Comment (vicmgs) [reply assets/js/actions/alert.js:16]: router.ex will need to be changed too

File: assets/js/components/alerts/AlertIndexTable.jsx [hunk 2]
```diff
@@ -1,28 +1,86 @@
...
-  const [selectedRows, setSelectedRows] = useState([]);
-  const [allSelected, setAllSelected] = useState(false);
+export default (props) => {
+  const renderTrigger = (trigger) => {
+    switch (trigger) {
+      case 'device/group':
+        return (
+          <span><img src={DeviceGroupTriggerIcon} style={{ height: 20, marginRight: 5 }} />Device/Group</span>
+        );
+      case 'function':
```
Comment (vicmgs) [inline assets/js/components/alerts/AlertIndexTable.jsx:19]: do we have function triggers already for alerts? or is this just placeholder

File: assets/js/components/alerts/AlertsIndex.jsx [hunk 3]
```diff
@@ -1,18 +1,55 @@
...
+    setShowDeleteAlertModal(true);
+    setSelectedAlert(alert);
+  }
+
+  const closeDeleteAlertModal = () => {
+    setShowDeleteAlertModal(false);
+    setSelectedAlert(null);
+  }
+
+  console.log("rendering")
```
Comment (vicmgs) [inline assets/js/components/alerts/AlertsIndex.jsx:52]: log

File: lib/console_web/router.ex [hunk 4]
```diff
@@ -48,6 +48,8 @@ defmodule ConsoleWeb.Router do
     post "/labels/swap_label", LabelController, :swap_label
     post "/labels/update_notification_settings", LabelNotificationSettingsController, :update
     post "/labels/update_notification_webhooks", LabelNotificationWebhooksController, :update
+    resources "/alerts", AlertController, only: [:index, :show, :create]
```
Comment (vicmgs) [inline lib/console_web/router.ex:51]: only need :create and :delete here

File: lib/console_web/schema/schema.ex [hunk 5]
```diff
@@ -283,6 +303,10 @@ defmodule ConsoleWeb.Schema do
       resolve &Console.Labels.LabelResolver.find/2
     end
 
+    field :all_alerts, list_of(:alert) do
```
Comment (vicmgs) [inline lib/console_web/schema/schema.ex:306]: missing single device schema here?

Comment (vicmgs) [reply lib/console_web/schema/schema.ex:306]: nvm thought you added the show page already

**Comment by vicmgs** [comment]
great stuff! hooks on everything! I'll switch to it for multi-packet too

File: assets/js/actions/alert.js (hunk 1 above)
Comment (ambuznego) [reply assets/js/actions/alert.js:16]: you right you right

File: assets/js/components/alerts/AlertIndexTable.jsx (hunk 2 above)
Comment (ambuznego) [reply assets/js/components/alerts/AlertIndexTable.jsx:19]: we have none that have been defined by prod yet so just a placeholder for now

File: lib/console_web/schema/schema.ex (hunk 5 above)
Comment (ambuznego) [reply lib/console_web/schema/schema.ex:306]: not yet hehe

**Review by vicmgs — APPROVED** [review]
ship

**Pull request merged**